File: implementations/fsharp/src/Skight.AgentPlatform.FSharp/SystemInfoTool.fs
namespace Skight.AgentPlatform.FSharp

open System
open System.IO
open System.Runtime.InteropServices
open System.Text.Json
open System.Threading.Tasks

module SystemInfoTool =

    let getSystemInfoAsync (workspaceRoot: string) : Task<string> =
        Task.FromResult(
            try
                let info = {|
                    os = RuntimeInformation.OSDescription
                    architecture = RuntimeInformation.OSArchitecture.ToString()
                    framework = RuntimeInformation.FrameworkDescription
                    machine = Environment.MachineName
                    user = Environment.UserName
                    processors = Environment.ProcessorCount
                    workspace = workspaceRoot
                    currentDir = Directory.GetCurrentDirectory()
                |}

                let options = JsonSerializerOptions(WriteIndented = true)
                JsonSerializer.Serialize(info, options)
            with ex -> sprintf "Error fetching system info: %s" ex.Message
        )

File: implementations/csharp/src/Skight.AgentPlatform/SystemInfoTool.cs
using System;
using System.IO;
using System.Runtime.InteropServices;
using System.Text.Json;
using System.Threading.Tasks;

namespace Skight.AgentPlatform
{
    public static class SystemInfoTool
    {
        public static Task<string> GetSystemInfoAsync(string workspaceRoot)
        {
            try
            {
                var info = new
                {
                    os = RuntimeInformation.OSDescription,
                    architecture = RuntimeInformation.OSArchitecture.ToString(),
                    framework = RuntimeInformation.FrameworkDescription,
                    machine = Environment.MachineName,
                    user = Environment.UserName,
                    processors = Environment.ProcessorCount,
                    workspace = workspaceRoot,
                    currentDir = Directory.GetCurrentDirectory()
                };

                return Task.FromResult(JsonSerializer.Serialize(info, new JsonSerializerOptions { WriteIndented = true }));
            }
            catch (Exception ex)
            {
                return Task.FromResult($"Error fetching system info: {ex.Message}");
            }
        }
    }
}
